packages/router-api/src/primitives.rs

```diff
@@ -114,12 +114,33 @@ impl TryFrom<String> for Address {
     }
 }
 
+impl Address {
+    pub const fn is_address(value: &str) -> bool {
+        !value.is_empty() && !contains!(value, FIELD_DELIMITER)
+    }
+}
+
 impl std::fmt::Display for Address {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "{}", *self.0)
     }
 }
 
+#[macro_export]
+macro_rules! address {
```

Comment:
update tests to use this?

Reply:
yeah, that's a massive change though, so I didn't want to do that at the same time as introducing it